client/core/trade.go

```diff
@@ -1386,10 +1386,13 @@ func (t *trackedTrade) checkSwapFeeConfirms(match *matchTracker) bool {
 		// Confirmed will be set in the db.
 		return true
 	}
+	// Waiting until the swap is definitely confirmed in order to not
+	// keep calling the fee checker before the swap is confirmed.
+	mine, _ := match.confirms()
 	if match.Side == order.Maker {
-		return match.Status >= order.MakerSwapCast
+		return match.Status > order.MakerSwapCast || (match.Status == order.MakerSwapCast && mine > 1)
```

Comment:
Why not...

```Go
return match.Status >= order.MakerSwapCast && mine > 0
```

Reply:
Unless I'm mistaken the `confirms()` value is all that matters now.  If it's confirmed, does it matter what the match status is?

Reply:
The reason I did it like this is the confirmation updates seemed very irregular using the MultiRPC. On Etherscan I would see many confirmations and it would still show 0. I was worried that maybe in some case `setSwapConfirms` would not be called.

How about 

``` golang
    return match.Status > order.MakerSwapCast || mine > 0
```

Reply:
> On Etherscan I would see many confirmations while still showing 0 in the dex client. I was worried that maybe in some case `setSwapConfirms` would not be called.

In this case, surely the dynamic fee check would also failed to find it on the chain?